### ISSUE TITLE
Error on successful unmount

### ISSUE THREAD

**Issue by jhermsmeier** [resin-io-modules/mountutils#30]
**OS:** Mac OS Sierra

**NOTE:** All of these were run with the device plugged in and mounted:

```js
// 1st: success
> mountUtils.unmountDisk('/dev/disk2',(...argv) => console.log(argv))
< []
```
```js
// 2nd: obvious failure
> mountUtils.unmountDisk('/dev/rdisk2',(...argv) => console.log(argv))
< [Error: Unmount failed, invalid drive at <anonymous>:1:12]
```
```js
// 3rd: success, but error returned
> mountUtils.unmountDisk('/dev/disk2',(...argv) => console.log(argv))
< [Error: Unmount failed, invalid drive at <anonymous>:1:12]
```

**Comment by lurch** [comment]
Could you clarify? Did you re-mount (or re-plug) the disk between each of those attempts? Or something else? What does `// success` mean here? ;-)

**Comment by jhermsmeier** [comment]
> Did you re-mount (or re-plug) the disk between each of those attempts?

Yes, exactly – I thought "All of these were run with the device plugged in and mounted" was clear...

> What does // success mean here?

Success means the device was unmounted by the operating system, where the third run resulted in an unmounted device, but still returned an error even though there wasn't one.

**Comment by lurch** [comment]
> Yes, exactly – I thought "All of these were run with the device plugged in and mounted" was clear...

Sorry, no idea how I missed that!! Must have been temporary blindness... ;-)

Is this reproducible, or was it a one-off? I've never looked into the OSX unmounting code, but does "obvious failure" mean that you _expected_ it to not work? (and if so, why? Just out of curiosity)

**Comment by jhermsmeier** [comment]
Closed in #38
